mne/io/egi/egimff.py

```diff
@@ -253,21 +254,57 @@ def _read_locs(filepath, chs, egi_info):
     """Read channel locations."""
     fname = op.join(filepath, 'coordinates.xml')
     if not op.exists(fname):
-        return chs
+        return chs, []
+    reference_names = ('VREF', 'Vertex Reference')
+    dig_kind_map = {
+        '': FIFF.FIFFV_POINT_EEG,
+        'VREF': FIFF.FIFFV_POINT_EEG,
+        'Vertex Reference': FIFF.FIFFV_POINT_EEG,
+        'Left periauricular point': FIFF.FIFFV_POINT_CARDINAL,
+        'Right periauricular point': FIFF.FIFFV_POINT_CARDINAL,
+        'Nasion': FIFF.FIFFV_POINT_CARDINAL,
+    }
+    dig_ident_map = {
+        'Left periauricular point': FIFF.FIFFV_POINT_LPA,
+        'Right periauricular point': FIFF.FIFFV_POINT_RPA,
+        'Nasion': FIFF.FIFFV_POINT_NASION,
+    }
     numbers = np.array(egi_info['numbers'])
     coordinates = parse(fname)
     sensors = coordinates.getElementsByTagName('sensor')
+    dig_points = []
+    dig_reference = None
     for sensor in sensors:
+        name_element = sensor.getElementsByTagName('name')[0].firstChild
+        name = '' if name_element is None else name_element.data
         nr = sensor.getElementsByTagName('number')[0].firstChild.data.encode()
-        id = np.where(numbers == nr)[0]
+        coords = [float(sensor.getElementsByTagName(coord)[0].firstChild.data)
+                  for coord in 'xyz']
+        loc = np.array(coords) / 100  # cm -> m
+        # create dig entry
+        kind = dig_kind_map[name]
+        if kind == FIFF.FIFFV_POINT_CARDINAL:
+            ident = dig_ident_map[name]
+        else:
+            ident = int(nr)
+        dig_point = DigPoint({'kind': kind, 'ident': ident, 'r': loc,
+                              'coord_frame': FIFF.FIFFV_COORD_HEAD})
+        if name in reference_names:
+            dig_reference = dig_point
+        else:
+            dig_points.append(dig_point)
+        # add location to channel entry
+        id = np.flatnonzero(numbers == nr)
         if len(id) == 0:
             continue
-        loc = chs[id[0]]['loc']
-        loc[0] = sensor.getElementsByTagName('x')[0].firstChild.data
-        loc[1] = sensor.getElementsByTagName('y')[0].firstChild.data
-        loc[2] = sensor.getElementsByTagName('z')[0].firstChild.data
-        loc /= 100.  # cm -> m
-    return chs
+        chs[id[0]]['loc'][:3] = loc
+    # Insert reference properly
+    if dig_reference is not None:
+        dig_points.insert(0, dig_reference)
```

Comment:
Based on our discussion I'd revert this insertion and just leave `ident` numbers 0 to N corresponding to E0-EN as you probably had it before. It breaks from our convention of putting ref first but I think for a good reason (there is actually an all-zero ref channel and it's at the end). Sorry for the back-and forth

Reply:
EGI actually starts numbering at `E1` (at least in the montages I've seen) so in the 128 channel net the reference is `E129`. Is it ok to have `dig` `ident` numbers 1-129 with 0 missing?

Reply:
Yeah that should be okay I think